### ISSUE TITLE
export-tar via http - need advice

### ISSUE THREAD

**Issue by knutov** [borgbackup/borg#4156]
As continue of #2492

borg 1.1.7, I want to be able download tar of snapshot via http with some access checks in http params via my cgi script.

The problem seems to be in buffering. In my setup I have: nginx -> uwsgi -> cgi script (in perl).

uswgi buffering is disabled in nginx.
I'm opening `borg export-tar` via pipe without buffering  (at least I think so):
```perl
    my $cmd = "LANG=en_US.UTF-8 /usr/local/bin/borg export-tar path-to-repo::snapshot -";
    my $pid = open(my $fh, '-|:unix', $cmd);
```
then read it and sending to uwsgi.

When http query made from local server or nearest server over gigabit network everything is fine and I'm able to download tar with size like 40GB.

But if connection speed is slow (100mbit/s for example) - I have error in uwsgi log like:
```
uwsgi_response_write_body_do() TIMEOUT !!!
[pid: 872636|app: -1|req: -1/5] 91.210.104.227 () {62 vars in 763 bytes} [Tue Nov  6 15:37:33 2018] GET /simple/test => generated 14713962496 bytes in 445521 msecs (HTTP/1.0 200) 4 headers in 152 bytes (109 switches on core 2)
```
after sending small part of data, like 1-5 GB, and uwsgi sends kill-like signal to my cgi script after that..

I googled and everywhere saw the solution is to disable buffering in nginx. I did it, nothing changes.
I googled more and start to open pipe in perl like popen ( `'-|:unix'` in my example) - didn't help.
The I tryed `stdbuf -i0 -o0 -e0 borg export-tar...` - didn't help too.

I will be glad to any advice, what I'm doing wrong?

**Comment by ThomasWaldmann** [comment]
No idea, but doesn't look like a borg problem.

**Comment by infectormp** [comment]
@knutov maybe you need try to set `uwsgi_socket_keepalive`? Available from nginx 1.15.6

**Comment by knutov** [comment]
just for history: ther problem was solved by `socket-timeout = 600` in uwsgi config, although we was unable to find/measure any real timeout.
